### ISSUE TITLE
Config file path:/home/sckj/yolov8/DeepStream-Yolo-master/config_infer_primary_ yolov8.txt,NvDsInfer Error:NVDSINFER CONFIG FAILED

### ISSUE THREAD

**Issue by 1354065425** [marcoslucianops/DeepStream-Yolo#395]
ferContextImpl::generateBackendContext()<nvdsinfer_context_impl.cpp:2029> [UID
= 1]: build backend context failed
0:17:27.919840104 15844 0xaaaaefab6300 ERROR
nvinfer gstnvinfer.c
pp:674:gst_nvinfer_logger:<primary_gie> NvDsInferContext[UID 1]: Error in NvDsIn
ferContextImpl::initialize() <nvdsinfer_context_impl.cpp:1266> [UID = 1]: genera
te backend failed, check config file settings
0:17:27.919939849 15844 0xaaaaefab6300 WARN
nvinfer gstnvinfer.c
pp:888:gst_nvinfer_start:<primary_gie> error: Failed to create NvDsInferContext
instance
0:17:27.919998569 15844 0xaaaaefab6300 WARN
nvinfer gstnvinfer.c
pp:888:gst_nvinfer_start:<primary_gie> error: Config file path: /home/sckj/yolov
8/Deepstream-Yolo-master/config_infer_primary_yoloV8.txt，NvDsInfer Error: NVDSI
NFER CONFIG FAILED
** ERROR: <main:716>: Failed to set pipeline to PAUSED
Quitting

**Comment by 1354065425** [comment]
![image](https://github.com/marcoslucianops/DeepStream-Yolo/assets/57380471/e2901c6a-c66b-4cc5-9c81-b49eda2c4e35)

**Comment by Today-fine** [comment]
Same error with my model using V5, working on it

**Comment by 1354065425** [comment]
my model is yolov8，follow the guidance，but not work

**Comment by Today-fine** [comment]
> 

I have no problem switching to Orin according to the official document, but I can’t switch to Xavier

**Comment by Today-fine** [comment]
> 我的型号是yolov8，按照指导操作，但是不行



> my model is yolov8，follow the guidance，but not work

I found the problem, unplugged the monitor, and then changed it to file output to generate the engine file, but my device has a problem and the power is cut off when playing the video, OP can try

**Comment by Today-fine** [comment]
> my model is yolov8，follow the guidance，but not work

I found the problem, unplugged the monitor, and then changed it to file output to generate the engine file, but my device has a problem and the power is cut off when playing the video, OP can try

**Comment by marcoslucianops** [comment]
> 0:17:27.919998569 15844 0xaaaaefab6300 WARN nvinfer gstnvinfer.cpp:888:gst_nvinfer_start:<primary_gie> error: Config file path: /home/sckj/yolov8/Deepstream-Yolo-master/config_infer_primary_yoloV8.txt，NvDsInfer Error: NVDSINFER CONFIG FAILED

Try to use the model exported without the `--dynamic` flag.

**Comment by 1354065425** [comment]
> > 0:17:27.919998569 15844 0xaaaaefab6300 WARN nvinfer gstnvinfer.cpp:888:gst_nvinfer_start:<primary_gie> error: Config file path: /home/sckj/yolov8/Deepstream-Yolo-master/config_infer_primary_yoloV8.txt，NvDsInfer Error: NVDSINFER CONFIG FAILED
> 
> Try to use the model exported without the `--dynamic` flag.

ok, the model exported without the --dynamic flag, but not detection

**Comment by 1354065425** [comment]
![image](https://github.com/marcoslucianops/DeepStream-Yolo/assets/57380471/c00a74ee-c290-472f-9d74-f2b949d613a7)
`python3 gen_wts_yoloV8.py -w yolov8s.pt -s 640`
`cp labels.txt Deepstream-yolo`
`cp yolov8s.onnx Deepstream-yolo`
`cd Deepstream-yolo && CUDA_VER=11.4 make -C nvdsinfer_custom_impl_Yolo`
`deepstream-app -c deepstream_app_config.txt`
for model_b1_gpu0_fp32.engine, but not detect this videos

**Comment by marcoslucianops** [comment]
The wts method is outdated and isn't supported anymore. Please check the new docs for export method.

**Comment by 1354065425** [comment]
> The wts method is outdated and isn't supported anymore. Please check the new docs for export method.
i use the command python3 export_yoloV8.py -w yolov8s.pt -s 640

**Comment by 1354065425** [comment]
> The wts method is outdated and isn't supported anymore. Please check the new docs for export method.

i use the command python3 export_yoloV8.py -w yolov8s.pt -s 640 not wts method

**Comment by marcoslucianops** [comment]
The last comment was `python3 gen_wts_yoloV8.py -w yolov8s.pt -s 640`. That's why I said that.

**Comment by marcoslucianops** [comment]
Please send the output from the terminal when you run the `deepstream-app`.

**Comment by 1354065425** [comment]
> Please send here the output from the terminal when you run the `deepstream-app`.

![68b6afd62dd4ad2609d31787e96ed1b](https://github.com/marcoslucianops/DeepStream-Yolo/assets/57380471/42a240df-783d-4b35-81f4-354cce06fc12)

**Comment by marcoslucianops** [comment]
Please send the full output. I would like to see the output layers/shape.

**Comment by 1354065425** [comment]
> Please send the full output. I would like to see the output layers/shapes.请发送完整输出。我想看看输出的图层/形状。

![049f6e16ee17c2b4132d73d49f9d614](https://github.com/marcoslucianops/DeepStream-Yolo/assets/57380471/75773090-5c8e-4a61-b484-3b8628def1a8)

**Comment by 1354065425** [comment]
> Please send the full output. I would like to see the output layers/shapes

![049f6e16ee17c2b4132d73d49f9d614](https://github.com/marcoslucianops/DeepStream-Yolo/assets/57380471/3ea67421-39ff-4516-a12f-ff2aae724905)

**Comment by 1354065425** [comment]
> It seems ok. Can you send the deepstream_app_config.txt and the config_infer_primary_yoloV8.txt files?

Okay, thank you for your reply
This is my deepstream_app_config.txt and config_infer_primary_yoloV8.txt
[config_infer_primary_yoloV8.txt](https://github.com/marcoslucianops/DeepStream-Yolo/files/12034631/
[deepstream_app_config.txt](https://github.com/marcoslucianops/DeepStream-Yolo/files/12034632/deepstream_app_config.txt)
config_infer_primary_yoloV8.txt)

**Comment by 1354065425** [comment]
> It seems ok. Can you send the deepstream_app_config.txt and the config_infer_primary_yoloV8.txt files?

Okay, thank you for your reply
This is my deepstream_app_config.txt and config_infer_primary_yoloV8.txt
[config_infer_primary_yoloV8.txt](https://github.com/marcoslucianops/DeepStream-Yolo/files/12034636/config_infer_primary_yoloV8.txt)
[deepstream_app_config.txt](https://github.com/marcoslucianops/DeepStream-Yolo/files/12034638/deepstream_app_config.txt)

**Comment by marcoslucianops** [comment]
Please change

```
#net-scale-factor=0.0039215697906911373
```

to

```
net-scale-factor=0.0039215697906911373
```

in the `config_infer_primary_yoloV8.txt` file.

**Comment by 1354065425** [comment]
> Please change
> 
> ```
> #net-scale-factor=0.0039215697906911373
> ```
> 
> to
> 
> ```
> net-scale-factor=0.0039215697906911373
> ```
> 
> in the `config_infer_primary_yoloV8.txt` file.

no decetion

**Comment by marcoslucianops** [comment]
Can you test the official YOLOv8s or other variant with uncomented `net-scale-factor=0.0039215697906911373`? Did you add normalization to your model in the training?

**Comment by 1354065425** [comment]
> Can you test the official YOLOv8s or other variant with uncomented `net-scale-factor=0.0039215697906911373`? Did you add normalization to your model in the training?

Okay, I'll try yolov8s now

**Comment by 1354065425** [comment]
> Can you test the official YOLOv8s or other variant with uncomented `net-scale-factor=0.0039215697906911373`? Did you add normalization to your model in the training?

The steps are all based on your document, but even if it is not recognized, the video can still be played out

**Comment by marcoslucianops** [comment]
The YOLOv8 needs the `net-scale-factor=0.0039215697906911373`. You need to change it only if your model is trained with image normalization.

**Comment by 1354065425** [comment]
> The YOLOv8 needs the `net-scale-factor=0.0039215697906911373`. You need to change it only if your model is trained with image normalization.
![3e3210bc903f251dc8698fd2f26fe4c](https://github.com/marcoslucianops/DeepStream-Yolo/assets/57380471/85d52eae-9524-48c1-a211-65d2695a9e3e)
yolov8s Still the Same

**Comment by marcoslucianops** [comment]
Did you delete the old engine file to run the YOLOv8s?

**Comment by 1354065425** [comment]
> Did you delete the old engine file to run the YOLOv8s?

yes，I removed previous engine

**Comment by 1354065425** [comment]
> Did you delete the old engine file to run the YOLOv8s?

I still haven't found where the problem lies

**Comment by marcoslucianops** [comment]
Hi, send the ONNX model to my email please.

**Comment by 1354065425** [comment]
> Hi, send the ONNX model to my email please.

Okay, I sent you an email,My email name is wangjin0992@gmail.com

**Comment by 1354065425** [comment]
> Hi, send the ONNX model to my email please.

Hey, brother, I tried using YOLOV7 and it's okay. It can recognize it normally, but YOLOV8 doesn't recognize it. Is the network structure of YOLOV8 different? Is the head and network separate?

**Comment by marcoslucianops** [comment]
What's your PyTorch version?

**Comment by 1354065425** [comment]
> What's your PyTorch version?

torch version is 1.14 
![image](https://github.com/marcoslucianops/DeepStream-Yolo/assets/57380471/2000c613-cc14-440e-863f-725b6aad6c2d)

**Comment by marcoslucianops** [comment]
It should work with PyTorch 1.14. The network is different, but the code is ready to work with the YOLOv7 and YOLOv8 when exported with the export file available on this repo.
